src/components/FilterAnnotMondal/FilterAnnotModal.js

```diff
@@ -0,0 +1,152 @@
+import React, { useState, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+
+import defaultTool from 'constants/defaultTool';
+import core from 'core';
+import actions from 'actions';
+import selectors from 'selectors';
+import Choice from 'components/Choice';
+
+import { Swipeable } from 'react-swipeable';
+
+import './FilterAnnotModal.scss';
+
+const FilterAnnotModal = () => {
+  const [isDisabled, isOpen] = useSelector(state => [
+    selectors.isElementDisabled(state, 'filterModal'),
+    selectors.isElementOpen(state, 'filterModal'),
+  ]);
+  const [t] = useTranslation();
+  const dispatch = useDispatch();
+
+  const [authors, setAuthors] = useState([]);
+  const [annotTypes, setAnnotTypes] = useState([]);
+
+  const [authorFilter, setAuthorFilter] = useState([]);
+  const [typesFilter, setTypesFilter] = useState([]);
+
+  const filterApply = () => {
+    dispatch(
+      actions.setCustomNoteFilter(annot => {
+        return (
+          typesFilter.includes(annot.Subject.toLowerCase().replace(/\s/g, '')) && authorFilter.includes(annot.Author)
+        );
+      }),
+    );
+    closeModal();
+  };
+
+  const filterClear = () => {
+    dispatch(
+      actions.setCustomNoteFilter(annot => {
+        return true;
+      }),
+    );
+    setAuthorFilter([]);
+    setTypesFilter([]);
+    closeModal();
+  };
+
+  const closeModal = () => {
+    dispatch(actions.closeElement('filterModal'));
+    core.setToolMode(defaultTool);
+  };
+
+  useEffect(() => {
+    const annots = core.getAnnotationsList();
+    let authorsToBeAdded = new Set();
+    let annotTypesToBeAdded = new Set();
+    annots.forEach(annot => {
+      authorsToBeAdded.add(core.getDisplayAuthor(annot));
+      annotTypesToBeAdded.add(annot.Subject.toLowerCase().replace(/\s/g, ''));
```

Comment:
I found that Link annotations don't have `Subject`, so opening the filter modal when a document has link annotations will blow up the app.